### PR TITLE
Work around for showAd

### LINKED ISSUE #36: Exception 'Only one `showAd` can be called at once' was throw
Could not catch the error while calling twice.
```
InterstitialAdManager.showAd('1220388591362840_1221348381266861')
      .then(didClick => console.log('click'))
      .catch(error => console.log(error));
```

simulator screen shot 7 feb 2017 10 23 45 pm

### PR DESCRIPTION
**Pull request by jakubsta**
This PR fixes #36

### CODE REVIEW

File: src/InterstitialAdManager.js [hunk 1]
```diff
@@ -12,11 +12,19 @@ import { NativeModules } from 'react-native';
 
 const { CTKInterstitialAdManager } = NativeModules;
 
+let pendingPromise: ?Promise<boolean> = null;
 export default {
   /**
    * Shows interstitial ad for a given placementId
    */
   showAd(placementId: string): Promise<boolean> {
-    return CTKInterstitialAdManager.showAd(placementId);
```
Comment (satya164) [inline src/InterstitialAdManager.js]: Maybe we can just reject the promise when another ad is already shown

Comment (grabbou) [inline src/InterstitialAdManager.js]: We can `reject` it from native instead of having invariant?

Comment (jakubsta) [inline src/InterstitialAdManager.js]: Sure, but isn't it better to quit as fast as we only can?

Comment (grabbou) [inline src/InterstitialAdManager.js]: You are right, but I think the perf. differences are minimal. I imagine we could do it here and remove keeping a reference to it entirely on the backend side. I wanted to keep the JS API surface minimal, with exposing only necessary set of flow types.

**Comment by jakubsta** [comment]
@grabbou @satya164 I moved promise rejection to native.

Any better idea for `[[NSError alloc] initWithDomain:NSURLErrorDomain code:errno userInfo:nil];`?

**Comment by grabbou** [comment]
@jakubsta you can pass `nil` to last argument instead of creating `NSError`

**Pull request merged**